pyramid/tier_fs.go

```diff
@@ -363,13 +353,24 @@ type localFileRef struct {
 	fsRelativePath params.RelativePath
 }
 
+func (tfs *TierFS) storeLocalFile(rPath params.RelativePath, size int64, eviction params.Eviction) {
+	if !eviction.Store(rPath, size) {
+		// Rejected from cache, so deleted.  This is safe, but can only happen when
+		// the cache size was lowered -- so warn.
+		tfs.logger.WithFields(logging.Fields{
+			"path": rPath,
+			"size": size,
+		}).Warn("existing file immediately rejected from cache on startup (safe if cache size changed; continue)")
```

Comment:
This is the most common case where this could happen, but I'm not sure it's the only one.
For instance, if there were multiple deletions taking place while the system crashed.